packages/components/src/focal-point-picker/index.js

```diff
@@ -23,317 +26,170 @@ import {
-		);
-		const top = Math.max(
-			bounds.top,
-			Math.min( relativePoint.top, bounds.bottom )
+const arrowKeyList = [ 'ArrowDown', 'ArrowLeft', 'ArrowRight', 'ArrowUp' ];
+
+export default function FocalPointPicker( {
+	autoPlay = true,
+	className,
+	help,
+	label,
+	onChange,
+	onDrag,
+	onDragEnd,
+	onDragStart,
+	resolvePoint,
+	url,
+	value: valueProp = {
+		x: 0.5,
+		y: 0.5,
+	},
+} ) {
+	const [ point, setPoint ] = useState( valueProp );
+	// Tracks the internal point value when it’s yet to be propagated.
+	const refPoint = useRef();
+	const keepPoint = ( value ) => {
+		setPoint( value );
+		refPoint.current = value;
+	};
+	const sendPoint = ( value = refPoint.current ) => {
+		onChange?.( value );
+		refPoint.current = undefined;
+	};
+	// Uses the internal point if it is held or else the value from props.
+	const { x, y } = refPoint.current !== undefined ? point : valueProp;
+
+	const dragAreaRef = useRef();
+	const [ bounds, setBounds ] = useState( INITIAL_BOUNDS );
+	const refUpdateBounds = useRef( () => {
+		const { clientWidth: width, clientHeight: height } =
+			dragAreaRef.current;
+		// Falls back to initial bounds if the ref has no size. Since styles
+		// give the drag area dimensions even when the media has not loaded
+		// this should only happen in unit tests (jsdom).
+		setBounds(
+			width > 0 && height > 0 ? { width, height } : { ...INITIAL_BOUNDS }
 		);
-
-		let nextX =
-			( left - bounds.left ) /
-			( pickerDimensions.width - bounds.left * 2 );
-		let nextY =
-			( top - bounds.top ) / ( pickerDimensions.height - bounds.top * 2 );
-
-		// Enables holding shift to jump values by 10%
-		const step = byTenths ? 0.1 : 0.01;
-
-		nextX = roundClamp( nextX, 0, 1, step );
-		nextY = roundClamp( nextY, 0, 1, step );
-
-		return { x: nextX, y: nextY };
-	}
-	pickerDimensions() {
-		const containerNode = this.containerRef.current;
-
-		if ( ! containerNode ) {
-			return {
-				width: 0,
-				height: 0,
-				left: 0,
-				top: 0,
-			};
+	} );
+
+	useEffect( () => {
+		const updateBounds = refUpdateBounds.current;
+		const { defaultView } = dragAreaRef.current.ownerDocument;
+		defaultView.addEventListener( 'resize', updateBounds );
+		return () => defaultView.removeEventListener( 'resize', updateBounds );
+	}, [] );
+
+	// Updates the bounds to cover cases of unspecified media or load failures.
+	useIsomorphicLayoutEffect( () => void refUpdateBounds.current(), [] );
+
+	const { startDrag, endDrag, isDragging } = useDragging( {
+		onDragStart: ( event ) => {
+			dragAreaRef.current.focus();
+			const value = getValueWithinDragArea( event );
+			onDragStart?.( value, event );
+			keepPoint( value );
+		},
+		onDragMove: ( event ) => {
+			// Prevents text-selection when dragging.
+			event.preventDefault();
+			const value = getValueWithinDragArea( event );
+			onDrag?.( value, event );
+			keepPoint( value );
+		},
+		onDragEnd: ( event ) => {
+			onDragEnd?.( event );
+			sendPoint();
```

Comment:
I'm trying to understand the reason we need to maintain `point` and `refPoint` separately. Is it just for the `sendPoint()` on this line? In that case, couldn't we do a `const value = getValueWithinDragArea( event )` here as well?

Reply:
I'd wanted to optimize by not recalculating through `getValueWithinDragArea` because there is no change of pointer coordinates with this event. On the other hand, I also don't like having both ref and state (I'd previously had another way to avoid that but it also had its ugliness).

I'm open to changing this as you've posited. I can't imagine there would be a perceivable performance difference.

Reply:
In e0015da, I've removed `refPoint` and found it there was no need to recalculate the point `onDragEnd` either. Seems like in an earlier iteration I'd ran into some issue with the non-synchronous nature of the `point` and `isDragging` states but can't reproduce any of that now. Anyway, it's sweet to simplify this further!

Reply:
Makes sense to me, the last known value from the `dragmove` event should be already good.

Thank you for the explanation!